src/org/omegat/core/data/IProject.java

```diff
@@ -139,6 +139,14 @@ void compileProjectAndCommit(String sourcePattern, boolean doPostProcessing, boo
      */
     List<SourceTextEntry> getAllEntries();
 
+    /**
+     * Set translation for entry.
+     *
+     * @param ste source text entry.
+     * @param tmxEntry tmx entry.
+     */
+    void setTranslation(SourceTextEntry ste, TMXEntry tmxEntry);
```

Comment:
Should `setTranslation(SourceTextEntry ste, TMXEntry tmxEntry)` be `setTranslation(SourceTextEntry ste, ITMXEntry entry)`?

Reply:
If it only uses the read methods of tmxEntry I would  say yes. Instead, not sure